### PR TITLE
Improve SFP and DWDM

### PR DESCRIPTION
**Pull request by BeArchiTek**
- Transform SFP as Generic
- Add SFP Bidi specifics
- Add SFP DWDM specifics

### CODE REVIEW

File: extensions/dwdm/dwdm.yml [hunk 1]
```diff
@@ -2,18 +2,220 @@
...
+        optional: false
+        cardinality: one
+        kind: Parent
+
+  - name: DwdmChannel
     namespace: Dcim
     description: A DWDM channel with its wavelength and frequency.
     label: DWDM Channel
     icon: game-icons:laser-warning
+    menu_placement: DcimDwdmDevice
```
Comment (BeArchiTek) [inline extensions/dwdm/dwdm.yml:210]: I would put those "DWDM Device" under Device instead and let the DwdmChannel by itself or in Other

File: extensions/dwdm/dwdm.yml [hunk 2]
```diff
@@ -2,18 +2,220 @@
...
+          # - name: URM_P8
+          #   label: URM-P8
+          #   description: "Enhanced version of URM connectors with higher protection."
+          # - name: Splice
+          #   label: Splice
+          #   description: "Permanent fiber connection method where two fiber ends are fused."
+
 nodes:
-  - name: DWDMChannel
+  - name: DwdmDevice
```
Comment (BeArchiTek) [inline extensions/dwdm/dwdm.yml:110]: Should we just call it a MUX / DEMUX ?

File: extensions/dwdm/dwdm.yml [hunk 3]
```diff
@@ -2,18 +2,220 @@
...
+    order_by:
+      - name__value
+    display_labels:
+      - name__value
+    inherit_from:
+      - DcimPhysicalDevice
+    attributes:
+      - name: name
+        kind: Text
+        unique: true # TODO: maybe combine location / name
```
Comment (BeArchiTek) [inline extensions/dwdm/dwdm.yml:126]: Should not be unique.

File: extensions/dwdm/dwdm.yml (hunk 2 above)
Comment (BaptisteGi) [reply extensions/dwdm/dwdm.yml:110]: OADM (Optical Add Drop Multiplexer) is apparently the exact name ...

Should we call that "Optical Multiplexer" (rip demux) and put the entire name in the description or something?

Comment (BeArchiTek) [reply extensions/dwdm/dwdm.yml:110]: yep let's do that

**Comment by BaptisteGi** [comment]
I adjusted a few things around DWDM to make it generic enough to support CWDM if needed

When it's good to you I'll rename it to WDM
- [ ] Rename DWDM to WDM

**Comment by BeArchiTek** [comment]
All good for me

**Pull request merged**